sycl/include/sycl/ext/oneapi/properties/property_utils.hpp

```diff
@@ -305,6 +305,37 @@ struct SizeListToStrHelper<SizeList<>, CharList<>> : CharsToStr<> {};
 template <size_t... Sizes>
 struct SizeListToStr : SizeListToStrHelper<SizeList<Sizes...>, CharList<>> {};
 
+//******************************************************************************
+// Property mutual exclusivity
+//******************************************************************************
+
+// Specializations of the following trait should not consider itself a
+// conflicting property.
+template <typename PropKey, typename Properties>
+struct ConflictingProperties : std::false_type {};
+
+template <typename Properties, typename T>
+struct NoConflictingPropertiesHelper {};
+
+template <typename Properties, typename... Ts>
+struct NoConflictingPropertiesHelper<Properties, std::tuple<Ts...>>
+    : std::true_type {};
+
+template <typename Properties, typename T, typename... Ts>
+struct NoConflictingPropertiesHelper<Properties, std::tuple<T, Ts...>>
+    : NoConflictingPropertiesHelper<Properties, std::tuple<Ts...>> {};
+
+template <typename Properties, typename... Rest, typename PropT,
+          typename... PropValuesTs>
+struct NoConflictingPropertiesHelper<
+    Properties, std::tuple<property_value<PropT, PropValuesTs...>, Rest...>>
```

Comment:
Do we actually need to pack it in a tuple?

Reply:
I see! Tuple is fine. We could do some unpacking, but it shouldn't be a problem either way.